### PR TITLE
fix: vendor nconf to remove vulnerable yargs version

### PR DESCRIPTION
**Pull request by JackuB**
BREAKING CHANGE: vendoring nconf module and swapping its yargs dependency for minimist
_Might_ influence argv parsing.

Vendored nconf and transformed it to typescript and ES6 export/import syntax for better compatibility. Swapped yargs with minimist in the vendored nconf, trying to keep the full compatibility.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,12 @@
     "test": "node test/*.test.js --afoo --azoo='true' | tap-spec",
     "format": "prettier --write '{lib,test}/**/*.?s'"
   },
+  "files": [
+    "dist",
+    "lib",
```
Comment (darscan) [inline package.json:16]: why `lib`?

File: package.json [hunk 2]
```diff
@@ -11,6 +11,12 @@
...
+    "LICENSE",
```
Comment (darscan) [inline package.json:17]: these are included by default: https://docs.npmjs.com/cli/v6/configuring-npm/package-json#files

File: package.json (hunk 1 above)
Comment (JackuB) [reply package.json:16]: Wanted to explicitly keep the nconf LICENSE in the bundle and its place in the `/lib` near its source files seems like the best place

File: package.json (hunk 2 above)
Comment (JackuB) [reply package.json:17]: TIL

File: package.json (hunk 1 above)
Comment (JackuB) [reply package.json:16]: With `/lib`

```
npm notice package size:  34.8 kB
npm notice unpacked size: 149.5 kB
```

Without `/lib`

```
npm notice package size:  22.8 kB
npm notice unpacked size: 97.8 kB
```

**Review by darscan — APPROVED** [review]

**Pull request merged**